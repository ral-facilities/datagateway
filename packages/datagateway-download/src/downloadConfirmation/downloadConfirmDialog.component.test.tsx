import type { RenderResult } from '@testing-library/react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { UserEvent } from '@testing-library/user-event/dist/types/setup';
import * as React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { DownloadSettingsContext } from '../ConfigProvider';
import {
  downloadPreparedCart,
  getDownload,
  getDownloadTypeStatus,
  submitCart,
} from '../downloadApi';
import { mockedSettings } from '../testData';
import DownloadConfirmDialog from './downloadConfirmDialog.component';

jest.mock('../downloadApi');
jest.mock('datagateway-common', () => {
  const originalModule = jest.requireActual('datagateway-common');

  return {
    __esModule: true,
    ...originalModule,
    handleICATError: jest.fn(),
  };
});

const createTestQueryClient = (): QueryClient =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

const renderWrapper = (
  size: number,
  isTwoLevel: boolean,
  open: boolean
): RenderResult =>
  render(
    <QueryClientProvider client={createTestQueryClient()}>
      <DownloadSettingsContext.Provider value={mockedSettings}>
        <DownloadConfirmDialog
          totalSize={size}
          isTwoLevel={isTwoLevel}
          open={open}
          redirectToStatusTab={jest.fn()}
          setClose={jest.fn()}
          clearCart={jest.fn()}
        />
      </DownloadSettingsContext.Provider>
    </QueryClientProvider>
  );

describe('DownloadConfirmDialog', () => {
  let user: UserEvent;

  beforeEach(() => {
    user = userEvent.setup();
    // Cannot mock to epoch time as Britain adopted BST permanently from 1968
    // to 1971, so snapshot will be an hour out depending on the date locale.
    global.Date.now = jest.fn(() => new Date(2020, 0, 1, 1, 1, 1).getTime());

    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) =>
      Promise.resolve({
        type,
        disabled: false,
        message: '',
      })
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', async () => {
    // Pass in a size of 100 bytes and for the dialog to be open when mounted.
    const wrapper = renderWrapper(100, false, true);

    expect(
      await wrapper.findByLabelText('downloadConfirmDialog.dialog_arialabel')
    ).toMatchSnapshot();
  });

  it('should not load the download speed/time table when isTwoLevel is true', async () => {
    // Set isTwoLevel to true as a prop.
    const wrapper = renderWrapper(100, true, true);

    expect(
      await wrapper.findByLabelText('downloadConfirmDialog.dialog_arialabel')
    ).toMatchSnapshot();
  });

  it('should prevent a download if a selected access method is disabled', async () => {
    // disable https method for testing
    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) =>
      Promise.resolve(
        type === 'https'
          ? {
              type,
              disabled: true,
              message: 'Disabled for testing',
            }
          : { type, disabled: false, message: '' }
      )
    );

    renderWrapper(100, false, true);

    await user.selectOptions(
      await screen.findByLabelText('downloadConfirmDialog.access_method_label'),
      ['https']
    );

    expect(
      await screen.findByText('downloadConfirmDialog.download')
    ).toBeDisabled();

    await user.selectOptions(
      await screen.findByLabelText('downloadConfirmDialog.access_method_label'),
      ['globus']
    );

    expect(
      await screen.findByText('downloadConfirmDialog.download')
    ).toBeEnabled();
  });

  it('should prevent a download if there are no available access methods', async () => {
    // Override default requests and return access method status'
    // as being disabled for both access methods.
    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) => ({
      type,
      disabled: true,
      message: 'disabled for testing',
    }));

    renderWrapper(100, false, true);

    expect(
      await screen.findByRole('button', {
        name: 'downloadConfirmDialog.download',
      })
    ).toBeDisabled();
  });

  it('should prevent download of an access method where the status was not fetched', async () => {
    // Return a response where one of the status requests has not been successful.
    (getDownloadTypeStatus as jest.Mock)
      .mockImplementationOnce((type, _) =>
        Promise.resolve({ type, disabled: true, message: '' })
      )
      .mockRejectedValueOnce({
        message: 'Test error message',
      });

    renderWrapper(100, false, true);

    await waitFor(() => {
      expect(screen.queryByRole('option', { name: 'GLOBUS' })).toBeNull();
      expect(screen.getByRole('option', { name: 'HTTPS' })).toBeInTheDocument();
    });
  });

  it('should show successful view when download is successful', async () => {
    (submitCart as jest.Mock).mockResolvedValue(123);
    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) =>
      Promise.resolve({
        type,
        disabled: false,
        message: '',
      })
    );

    renderWrapper(100, true, true);
    // input an email
    await user.type(
      await screen.findByLabelText('downloadConfirmDialog.email_label'),
      'test@email.com'
    );
    // input a download name
    await user.type(
      await screen.findByLabelText('downloadConfirmDialog.download_name_label'),
      'custom download name'
    );
    // click on download button to begin download
    await user.click(await screen.findByText('downloadConfirmDialog.download'));

    // should not show error
    await waitFor(() => {
      expect(
        screen.queryByText('Your download request was unsuccessful', {
          exact: false,
        })
      ).toBeNull();
    });

    expect(submitCart).toHaveBeenCalledWith(
      'https',
      'test@email.com',
      'custom download name',
      {
        facilityName: mockedSettings.facilityName,
        downloadApiUrl: mockedSettings.downloadApiUrl,
      },
      undefined
    );
    // should show success message
    expect(
      await screen.findByText('downloadConfirmDialog.download_success')
    ).toBeInTheDocument();
    // should show confirmation email address
    expect(await screen.findByText('test@email.com')).toBeInTheDocument();
    expect(await screen.findByText('custom download name')).toBeInTheDocument();
  });

  it('should download prepared cart upon successful submission of cart', async () => {
    (submitCart as jest.Mock).mockResolvedValue(123);
    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) =>
      Promise.resolve({
        type,
        disabled: false,
        message: '',
      })
    );
    (getDownload as jest.Mock).mockResolvedValue({
      preparedId: 1,
      fileName: 'test-file-name',
      status: 'COMPLETE',
    });

    renderWrapper(100, true, true);
    // click on download button to begin download
    await user.click(await screen.findByText('downloadConfirmDialog.download'));

    await waitFor(() => {
      expect(downloadPreparedCart).toHaveBeenCalledWith(1, 'test-file-name', {
        idsUrl: 'https://example.com/ids',
      });
    });
  });

  it('should disable download when no download method is available', async () => {
    (getDownloadTypeStatus as jest.Mock).mockRejectedValue({
      error: 'test error',
    });

    renderWrapper(100, true, true);

    expect(
      await screen.findByText(
        'downloadConfirmDialog.access_method_helpertext_all_disabled_error'
      )
    ).toBeInTheDocument();
    expect(
      await screen.findByRole('button', {
        name: 'downloadConfirmDialog.download',
      })
    ).toBeDisabled();
  });

  it('should show error when download has failed', async () => {
    (submitCart as jest.Mock).mockRejectedValue({
      message: 'error',
    });

    renderWrapper(100, true, true);
    // click on download button to begin download
    await user.click(await screen.findByText('downloadConfirmDialog.download'));

    // should not show success message
    await waitFor(() => {
      expect(
        screen.queryByText('downloadConfirmDialog.download_success')
      ).toBeNull();
    });
    // should show error
    expect(
      await screen.findByText('Your download request was unsuccessful', {
        exact: false,
      })
    ).toBeInTheDocument();
  });

  it('should prevent the submission of a download request with an invalid email', async () => {
    renderWrapper(100, false, true);

    const emailInput = await screen.findByLabelText(
      'downloadConfirmDialog.email_label'
    );

    // type in an incorrect email address
    await user.type(emailInput, 'cat-person@');

    expect(
      await screen.findByRole('button', {
        name: 'downloadConfirmDialog.download',
      })
    ).toBeDisabled();

    // Simulate correct email address entry.
    await user.clear(emailInput);
    await user.type(emailInput, 'cat@person.com');

    await waitFor(async () => {
      expect(
        await screen.findByText('downloadConfirmDialog.download')
      ).toBeEnabled();
    });

    // Simulate removing an email address completely,
    // resulting in emptying the text field.
    await user.clear(emailInput);

    await waitFor(async () => {
      expect(
        await screen.findByText('downloadConfirmDialog.download')
      ).toBeEnabled();
    });
  });

  it('should close the download Confirmation Dialog and successfully call the setClose function', async () => {
    const closeFunction = jest.fn();

    render(
      <QueryClientProvider client={new QueryClient()}>
        <DownloadSettingsContext.Provider value={mockedSettings}>
          <DownloadConfirmDialog
            totalSize={1}
            isTwoLevel={false}
            open={true}
            redirectToStatusTab={jest.fn()}
            setClose={closeFunction}
            clearCart={jest.fn()}
          />
        </DownloadSettingsContext.Provider>
      </QueryClientProvider>
    );

    await user.click(
      await screen.findByLabelText('downloadConfirmDialog.close_arialabel')
    );

    expect(closeFunction).toHaveBeenCalled();
  });
});

describe('DownloadConfirmDialog - renders the estimated download speed/time table with varying values', () => {
  const timeWrapper = (size: number): RenderResult =>
    render(
      <QueryClientProvider client={new QueryClient()}>
        <DownloadSettingsContext.Provider value={mockedSettings}>
          <DownloadConfirmDialog
            totalSize={size}
            isTwoLevel={false}
            open={true}
            setClose={jest.fn()}
            redirectToStatusTab={jest.fn()}
            clearCart={jest.fn()}
          />
        </DownloadSettingsContext.Provider>
      </QueryClientProvider>
    );

  beforeEach(() => {
    (getDownloadTypeStatus as jest.Mock).mockImplementation((type, _) =>
      Promise.resolve({
        type,
        disabled: false,
        message: '',
      })
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  // Calculate the file size required to reach the given download time (at 1 Mbps).
  const timeToSize = (
    seconds: number,
    minutes?: number,
    hours?: number,
    days?: number
  ): number => {
    // NOTE: For these tests to make it simple we use 1 Mbps for this.
    const downloadSpeed = 1; // Mbps

    // Get the all the time in seconds.
    const inSeconds =
      (days ? days * 86400 : 0) +
      (hours ? hours * 3600 : 0) +
      (minutes ? minutes * 60 : 0) +
      seconds;

    // Calculate final file size required (in bytes).
    const fileSize = inSeconds * (downloadSpeed / 8) * (1024 * 1024);

    return fileSize;
  };

  it('renders for multiple days, hours, minutes and seconds', async () => {
    // Test for 2 seconds, 2 minutes, 2 hours and 2 days.
    timeWrapper(timeToSize(2, 2, 2, 2));

    expect(
      await screen.findByText(
        '2 downloadConfirmDialog.day {count:2}, 2 downloadConfirmDialog.hour {count:2}, 2 downloadConfirmDialog.minute {count:2}, 2 downloadConfirmDialog.second {count:2}'
      )
    ).toBeInTheDocument();
  });

  it('renders for a single day, hour, minute and second', async () => {
    timeWrapper(timeToSize(1, 1, 1, 1));

    expect(
      await screen.findByText(
        '1 downloadConfirmDialog.day {count:1}, 1 downloadConfirmDialog.hour {count:1}, 1 downloadConfirmDialog.minute {count:1}, 1 downloadConfirmDialog.second {count:1}'
      )
    ).toBeInTheDocument();
  });

  describe('estimated download table renders for single time measurements', () => {
    it('renders for a single day', async () => {
      timeWrapper(timeToSize(0, 0, 0, 1));

      expect(
        await screen.findByText('1 downloadConfirmDialog.day {count:1}')
      ).toBeInTheDocument();
    });

    it('renders for a single hour', async () => {
      timeWrapper(timeToSize(0, 0, 1, 0));

      expect(
        await screen.findByText('1 downloadConfirmDialog.hour {count:1}')
      ).toBeInTheDocument();
    });

    it('renders for a single minute', async () => {
      timeWrapper(timeToSize(0, 1, 0, 0));

      expect(
        await screen.findByText('1 downloadConfirmDialog.minute {count:1}')
      ).toBeInTheDocument();
    });

    it('renders for a single second', async () => {
      timeWrapper(timeToSize(1, 0, 0, 0));

      expect(
        await screen.findByText('1 downloadConfirmDialog.second {count:1}')
      ).toBeInTheDocument();
    });
  });
});
