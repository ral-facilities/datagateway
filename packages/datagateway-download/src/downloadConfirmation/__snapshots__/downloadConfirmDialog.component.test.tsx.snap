// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadConfirmDialog renders correctly 1`] = `
<div
  aria-label="downloadConfirmDialog.dialog_arialabel"
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="mui-1"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <div>
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-140puxv-MuiTypography-root-MuiDialogTitle-root"
          id="mui-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1654shd-MuiTypography-root"
          >
            downloadConfirmDialog.dialog_title
          </p>
          <button
            aria-label="downloadConfirmDialog.close_arialabel"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1f5482o-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </h2>
        <div
          class="MuiDialogContent-root css-1738ltc-MuiDialogContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="confirm-download-name"
                  id="confirm-download-name-label"
                >
                  downloadConfirmDialog.download_name_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-describedby="confirm-download-name-helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="confirm-download-name"
                    maxlength="255"
                    placeholder="LILS_2020-1-1_1-1-1"
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-download-name-helper-text"
                >
                  downloadConfirmDialog.download_name_helpertext
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root css-pe0pe4-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="confirm-access-method"
                >
                  downloadConfirmDialog.access_method_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <select
                    aria-invalid="false"
                    class="MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input MuiInput-input css-1dmqq7i-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
                    id="confirm-access-method"
                    name="Access Method"
                  >
                    <option
                      aria-label="HTTPS"
                      id="confirm-access-method-https"
                      value="https"
                    >
                      HTTPS
                    </option>
                    <option
                      aria-label="Globus"
                      id="confirm-access-method-globus"
                      value="globus"
                    >
                      Globus
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-access-method-help"
                >
                  downloadConfirmDialog.access_method_helpertext
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <span
                style="padding-top: 20px;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    downloadConfirmDialog.access_method_info
                    :
                  </b>
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  id="confirm-access-method-https-description"
                >
                  Example description for 
                  <b>
                    HTTPS
                  </b>
                   access method.
                </p>
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                <b>
                  downloadConfirmDialog.download_size
                  :
                </b>
                 
                100 B
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                id="estimated-download-times"
              >
                downloadConfirmDialog.estimated_download_times
                :
              </p>
              <div
                class="css-bk1bta"
              >
                <table
                  aria-labelledby="estimated-download-times"
                  id="download-table"
                >
                  <tbody>
                    <tr>
                      <th>
                        1 Mbps
                      </th>
                      <th>
                        30 Mbps
                      </th>
                      <th>
                        100 Mbps
                      </th>
                    </tr>
                    <tr>
                      <td
                        id="download-table-one"
                      >
                        &lt; 1 downloadConfirmDialog.second {count:1}
                      </td>
                      <td
                        id="download-table-thirty"
                      >
                        &lt; 1 downloadConfirmDialog.second {count:1}
                      </td>
                      <td
                        id="download-table-hundred"
                      >
                        &lt; 1 downloadConfirmDialog.second {count:1}
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="confirm-download-email"
                  id="confirm-download-email-label"
                >
                  downloadConfirmDialog.email_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-describedby="confirm-download-email-helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="confirm-download-email"
                    maxlength="254"
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-download-email-helper-text"
                >
                  downloadConfirmDialog.email_help
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-17tk4tg-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            id="download-confirmation-download"
            tabindex="0"
            type="button"
          >
            downloadConfirmDialog.download
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`DownloadConfirmDialog should not load the download speed/time table when isTwoLevel is true 1`] = `
<div
  aria-label="downloadConfirmDialog.dialog_arialabel"
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="mui-2"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <div>
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-140puxv-MuiTypography-root-MuiDialogTitle-root"
          id="mui-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1654shd-MuiTypography-root"
          >
            downloadConfirmDialog.dialog_title
          </p>
          <button
            aria-label="downloadConfirmDialog.close_arialabel"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1f5482o-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </h2>
        <div
          class="MuiDialogContent-root css-1738ltc-MuiDialogContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="confirm-download-name"
                  id="confirm-download-name-label"
                >
                  downloadConfirmDialog.download_name_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-describedby="confirm-download-name-helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="confirm-download-name"
                    maxlength="255"
                    placeholder="LILS_2020-1-1_1-1-1"
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-download-name-helper-text"
                >
                  downloadConfirmDialog.download_name_helpertext
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root css-pe0pe4-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="confirm-access-method"
                >
                  downloadConfirmDialog.access_method_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <select
                    aria-invalid="false"
                    class="MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input MuiInput-input css-1dmqq7i-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
                    id="confirm-access-method"
                    name="Access Method"
                  >
                    <option
                      aria-label="HTTPS"
                      id="confirm-access-method-https"
                      value="https"
                    >
                      HTTPS
                    </option>
                    <option
                      aria-label="Globus"
                      id="confirm-access-method-globus"
                      value="globus"
                    >
                      Globus
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-access-method-help"
                >
                  downloadConfirmDialog.access_method_helpertext
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <span
                style="padding-top: 20px;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    downloadConfirmDialog.access_method_info
                    :
                  </b>
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  id="confirm-access-method-https-description"
                >
                  Example description for 
                  <b>
                    HTTPS
                  </b>
                   access method.
                </p>
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                <b>
                  downloadConfirmDialog.download_size
                  :
                </b>
                 
                100 B
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="confirm-download-email"
                  id="confirm-download-email-label"
                >
                  downloadConfirmDialog.email_label
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-describedby="confirm-download-email-helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="confirm-download-email"
                    maxlength="254"
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                  id="confirm-download-email-helper-text"
                >
                  downloadConfirmDialog.email_help
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-17tk4tg-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            id="download-confirmation-download"
            tabindex="0"
            type="button"
          >
            downloadConfirmDialog.download
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
